### DISCUSSION TITLE
May I ask if the TIF file of CMYK is converted to RGB in this way? The extracted png has color difference

### DISCUSSION THREAD

**Discussion by 291598912** [ImageMagick/ImageMagick#6318] [Digital Image Processing]
convert input.tif -profile cmyk.icc -profile sRGB2.icc 
[input.zip](https://github.com/ImageMagick/ImageMagick/files/11440616/input.zip)
%d.png
I found that extracting png in this way has a slight color difference

**Comment by 291598912** [comment]
convert intput.tif -profile cmyk.icc -profile sRGB.icc -colorspace sRGB %d.png

**Comment by snibgo** [comment]
Yes, that is reasonable, as the CMYK inputs do not have embedded ICC colour profiles.

I suggest using `magick` instead of `convert`.

If you use an `-intent` or `-black-point-compensation` before the second `-profile`, the result may have slightly different colours. See https://imagemagick.org/script/command-line-options.php#intensity and https://imagemagick.org/script/command-line-options.php#black-point-compensation .

**Reply by 291598912** [reply]
magick intput.tif -profile cmyk.icc -intent relative -profile sRGB.icc -colorspace sRGB %d.png
Is that so? There is still a color difference

**Reply by fmw42** [reply]
If your tif has dot gain, Imagemagick cannot handle that.  See the color settings in Photoshop.  Also a CMYK color profile may have a larger gamut than sRGB or even AdobeRGB. So you may be losing colors in the conversion, since PNG does not allow CMYK.  Snibgo can correct me as he knows more about this than I do.

**Reply by 291598912** [reply]
I see that the converted rgb values always differ by 5-6

**Reply by 291598912** [reply]
Even if there is no color difference in the png image, the tif of the final synthesized CMYK will still exhibit color difference

**Reply by fmw42** [reply]
5-6 is just computational accuracy and should be fine. That is very small.  Using compare -metric rmse, what value do you get in parentheses. Multiply by 100 to get the approximate percent difference.

**Comment by fmw42** [comment]
Snibgo, I think you accidentally linked to the wrong location. You linked to intensity and I think you wanted intent.  So the intent link would be https://imagemagick.org/script/command-line-options.php#intent

**Reply by 291598912** [reply]
Is the version of the ICC configuration file incorrect

**Comment by snibgo** [comment]
Oops, yes, thanks, fmw42.
